runbooks/how-to-create-server-regression-dashboards.md

```diff
@@ -0,0 +1,343 @@
+
+## FAQ
+
+<!-- This section should contain frequently-asked-questions as sub-headers with answers as sub-paragraphs. Since this is a living document, this section should be added to as needed. -->
+
+<!-- REPLACE ME when FAQs are added -->
+
+## Instructions
+
+1. You will need to navigate to the QuickSight AWS Service in order to start creating QuickSight Dashboards
+
+   1. Navigate to `aws.amazon.com` in any browser
+   2. Click the "Sign In" link in the navigation bar on the top right of the screen
+   3. Enter your AWS account credentials and sign-in
+   4. You should now have navigated to the "Console Home", click the search input in the navigation bar on the top left of the screen
+   5. Type "QuickSight" and click the QuickSight result that appears
+
+2. Every QuickSight dashboard requires a _Dataset_, which for the `server-regression` dashboards will be a `Custom SQL` query against Athena. As there will be three dashboards, there will be three datasets (one dataset for each dashboard)
+
+   1. Creating the `bfd-test-server-regression` dataset:
+
+      1. Navigate to "Datasets" in QuickSight (click on "Datasets" in the navigation tree on the left side of the page)
+      2. Click the "New dataset" button on the top-left of the page
+      3. The page that appears should list a large number of possible data sources. Find "Athena" in this list, and click on it
+      4. Enter "bfd-test-server-regression" for the "Data source name"
+      5. Choose `bfd` for the "Athena workgroup"
+      6. Click "Create data source"
+      7. Choose `AwsDataCatalog` as the "Catalog"
+      8. Choose `bfd-insights-bfd-test` as the "Database"
+      9. Choose `bfd_insights_bfd_test_server_regression" as the "Table"
+      10. Click "Use custom SQL"
+      11. In the first input box, enter "bfd-test-server-regression" (replacing "New custom SQL")
+      12. In the large text area, enter the following SQL query:
+
+      ```sql
+      select
+        totals.total_reqs_per_second,
+        totals.num_requests as total_requests,
+        totals.num_failures as total_failures,
+        totals.min_response_time as total_min_response_time,
+        totals.average_response_time as total_avg_response_time,
+        totals.median_response_time as total_median_response_time,
+        totals.response_time_percentiles."0.95" as total_95_percentile,
+        totals.response_time_percentiles."0.99" as total_99_percentile,
+        totals.response_time_percentiles."1.0" as total_100_percentile,
+        metadata.compare_result,
+        date_format(from_unixtime(metadata.timestamp, 'US/Eastern'), '%b-%e-%y %T EST') as date_time,
+        metadata.timestamp as raw_timestamp
+      from "bfd-insights-bfd-test"."bfd_insights_bfd_test_server_regression"
+      where
+        metadata.hash = 'ae770dea971f3be467a42717654a42c5c26c7da394f690c3db23e9aa4c4e2f92'
```

Comment:
There are really only two scenarios where the hash could change:

1. The parameters (runtime, user count, spawn rate) we pass to the regression suite from the deployment pipeline change
2. The code to generate the hash changes because we add an additional field or want to change the hashing algorithm, etc.

Right now I'm _fairly_ confident (1) is unlikely to change as we are seeing consistent results with minimal increase in deployment time. I'm even more confident (2) is unlikely to change as I can't really think of any additional metadata that Locust exposes that we would want to _include_ in the hash.

If we're concerned about this possibly changing in the future (and causing the runbook to become out-of-sync with the actual Datasets) we can definitely discuss creating a single source of truth for both the _actual_ QuickSight Datasets and the runbook.

Reply:
Might be worth mentioning those cases in the document somewhere at least, for posterity.

Reply:
Probably a good case for an "FAQ" answer. I think I'll add one for this case.